src/lib/birthday/utils.ts

```diff
@@ -104,20 +104,16 @@ export function compareDate(month: Month, day: number, { now = Date.now() }: Tim
 
 /**
  * Gets the current age from a date.
- * @param data The data to compare.
+ * @param date The date to compare.
  * @param options The options for the operation of this function.
- * @returns `null` if `data.year` is `null`, a number of years otherwise.
+ * @returns `null` if `date.year` is `null`, a number of years otherwise.
  */
-export function getAge(data: DateWithOptionalYear, { now = Date.now() }: TimeOptions = {}) {
-	if (data.year === null) return null;
-
-	// If the birthday has happened, we subtract the years by one, that way:
-	//
-	// * 2021/03/18 - 2020/05/10 = 0
-	// * 2021/03/18 - 2020/02/26 = 1
-	const extra = compareDate(data.month, data.day, { now }) === -1 ? 0 : -1;
-	const years = new Date(now).getUTCFullYear() - data.year;
-	return years + extra;
+export function getAge(date: DateWithOptionalYear, { now = Date.now() }: TimeOptions = {}) {
+	if (date.year === null) return null;
+
+	const birthDate = Date.UTC(date.year, date.month - 1, date.day);
+
+	return new Date(now - birthDate).getUTCFullYear() - new Date(0).getUTCFullYear();
```

Comment:
This is quite the interesting operation, however, we could make it slightly more efficient by not creating a constant over and over. Either create a constant outside this function (and not export), or hardcode `1970` since it's never changing.

Also, it'd be good to same some comment above to show why this works.

As a side note, does this function still work for people born before 1970?

Reply:
Yes, it does work for people born before 1970.

I will add some comments explaining the math of it and probably hardcode 1970

Reply:
> As a side note, does this function still work for people born before 1970?

Yes. since both `now` and `birthDate` are offset by 1970, `(now - birthDate).getUTCFullYear()` is equivalent to `age + 1970`, so we subtract 1970 and it works.

I'm not sure this needs a comment explaining the maths, honestly. Hardcoding 1970 would be a good idea, though, since I don't envision the UNIX epoch changing in our lifetimes.